CRM/Financeextras/Hook/BuildForm/RefundButton.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+use CRM_Financeextras_ExtensionUtil as E;
+
+/**
+ * Implement Refund Button if supported Refund by Payment Processor
+ */
+class CRM_Financeextras_Hook_BuildForm_RefundButton {
+
+
+  /**
+   * @var string
+   *   Path where template with javascript stored for adding new Refund Button.
+   */
+  private $templatePath;
+
+  /**
+   * @var \CRM_Contribute_Form_AdditionalPayment
+   *   Form object that is being altered.
+   */
+  private $form;
+
+  /**
+   * CRM_Financeextras_Hook_BuildForm_RefundButton constructor.
+   *
+   * @param \CRM_Contribute_Form_AdditionalPayment $form
+   */
+  public function __construct(&$form) {
+    $this->form = $form;
+    $this->templatePath = E::path() . '/templates/CRM/AdditionalPayment/Form/Payment';
+  }
+
+  /**
+   * Implements changes to form loaded in $this->form
+   */
+  public function buildForm() {
+    $this->addRefundButton();
+  }
+
+  /**
+   * Added Javascript in template to add the button for submit refund
+   */
+  private function addRefundButton() {
+    CRM_Core_Region::instance('page-body')->add([
+      'template' => "{$this->templatePath}/Refund.tpl",
```

Comment:
I checked the Refund.tpl again the name is quite vague and basically you just add a link or button, it should call RefundButton or SubmitCreditCardButton.tpl 

Furthormore, I don't think adding a tpl is appropriated here, since you don't have any markup in the file and the file only contains pure javascript.  You should add js file instead. See CiviCRM doc how it should be done. https://docs.civicrm.org/dev/en/latest/framework/resources/#javascript